stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +612,348 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
 
 	return fmt.Errorf("error waiting for metrics %v: %v", wantMetrics, ctx.Err())
 }
+
+// TestServerWithMetricsAndTraceOptions tests emitted metrics and traces from
+// OpenTelemetry instrumentation component. It then configures a system with a gRPC
+// Client and gRPC server with the OpenTelemetry Dial and Server Option configured
+// specifying all the metrics and traces provided by this package, and makes a Unary
+// RPC and a Streaming RPC. These two RPCs should cause certain recording for each
+// registered metric observed through a Manual Metrics Reader on the provided
+// OpenTelemetry SDK's Meter Provider. It also verifies the traces are recorded
+// correctly.
+func (s) TestServerWithMetricsAndTraceOptions(t *testing.T) {
+	// Create default metrics options
+	mo, reader := defaultMetricsOptions(t, nil)
+	// Create default trace options
+	to, exporter := defaultTraceOptions(t)
+
+	ss := setupStubServer(t, mo, to)
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	// Create a parent span for the client call
+	ctx, _ = otel.Tracer("grpc-open-telemetry").Start(ctx, "test-parent-span")
```

Comment:
why do we have to insert a test span before? The unary rpc call should add the span with the method name which is what we should verify?